src/main/java/org/jvnet/hudson/plugins/port_allocator/PortType.java

```diff
@@ -27,9 +27,7 @@
     public final String name;
 
     protected PortType(String name) {
-        // to avoid platform difference issue in case sensitivity of environment variables,
-        // always use uppser case.
-        this.name = name.toUpperCase();
+        this.name = name;
```

Comment:
Plugin must just save what user entered and then pass to environment. Current behaviour absolutely not obvious. I'm configuring job, entering variable "my_port", pressing save, then build is broken because code decided to do upper case. And this is really breaking change.